### PR TITLE
fix: various fixes

### PR DESCRIPTION
**Pull request by glmxndr**

### CODE REVIEW

File: applications/processor/download-datablocks/src/main/java/com/csgroup/s2pdgs/downloaddatablocks/DownloadDataBlocksProcessor.java [hunk 1]
```diff
@@ -167,7 +167,7 @@ private Catalog getCatalog(StationProduct product) {
         return catalogClientManager
             .monoWithCatalogClient(
                 getCatalog(product),
-                client -> client.updateAndRefresh(catalogIndex, product)
+                client -> client.update(catalogIndex, product)
```
Comment (glmxndr) [inline applications/processor/download-datablocks/src/main/java/com/csgroup/s2pdgs/downloaddatablocks/DownloadDataBlocksProcessor.java:170]: Index refreshes seemed to add a LOT of load on ES, which seems to have been the source of timeouts. I removed all the refreshes an ES has not shown timeouts.

File: applications/source/station-source/src/main/java/com/csgroup/s2pdgs/stationsource/StationProductsFinder.java [hunk 2]
```diff
@@ -88,16 +89,15 @@ public StationProductsFinder(
         return entriesFromBasePath()
             .flatMapMany(Flux::fromStream)
             .filterWhen(this::isSessionDir)
-            .filterWhen(this::isNotYetInCache)
```
Comment (glmxndr) [inline applications/source/station-source/src/main/java/com/csgroup/s2pdgs/stationsource/StationProductsFinder.java:91]: Caching session names before they are treated: caching too much too soon.

File: applications/source/station-source/src/main/java/com/csgroup/s2pdgs/stationsource/StationProductsFinder.java [hunk 3]
```diff
@@ -109,6 +109,7 @@ public StationProductsFinder(
             .filterWhen(this::isDSIB)
             .filterWhen(this::isNotTempFile)
             .flatMap(this::fromDSIB)
+            .filterWhen(this::isNotInCache)
```
Comment (glmxndr) [inline applications/source/station-source/src/main/java/com/csgroup/s2pdgs/stationsource/StationProductsFinder.java:112]: Ignore DSIBs in cache.

File: applications/source/station-source/src/main/java/com/csgroup/s2pdgs/stationsource/StationProductsFinder.java [hunk 4]
```diff
@@ -119,6 +120,7 @@ public StationProductsFinder(
             .filterWhen(this::isRawFile)
             .filterWhen(this::isNotTempFile)
             .flatMap(Products::from)
+            .filterWhen(this::isNotInCache)
```
Comment (glmxndr) [inline applications/source/station-source/src/main/java/com/csgroup/s2pdgs/stationsource/StationProductsFinder.java:123]: Ignore RAW files in cache.

File: applications/source/station-source/src/main/java/com/csgroup/s2pdgs/stationsource/StationProductsFinder.java [hunk 5]
```diff
@@ -60,8 +60,9 @@ public StationProductsFinder(
     public Flux<StationProduct> findProducts() {
         return refreshedIndex()
             .flatMapMany(unit -> findNewSessions())
-            .doOnNext(s -> log.info("findProducts: Session found: name={}", s.getName()))
+            .doOnNext(s -> log.trace("findProducts: Session found: name={}", s.getName()))
             .flatMapSequential(this::extractNewRawAndDSIB)
+            .doOnNext(sp -> sessionNameCache.putInCache(sp.getName()))
```
Comment (glmxndr) [inline applications/source/station-source/src/main/java/com/csgroup/s2pdgs/stationsource/StationProductsFinder.java:65]: Cache the DSIB/RAW file names (= not too much), after they have been detected to be eligible as messages, and just before rendering them as mesages (= not too soon).

File: common/common-utils/src/main/java/com/csgroup/s2pdgs/commonutils/DedupCache.java [hunk 6]
```diff
@@ -9,19 +9,33 @@
 
 import static com.csgroup.s2pdgs.app.processor.shellscript.fp.Unit.UNIT;
 
-@FunctionalInterface
 public interface DedupCache<K> {
 
-    Mono<Boolean> isNotYetInCache(K key);
-    default Mono<Boolean> isInCache(K key) {
-        return isNotYetInCache(key).map(b -> !b);
+    void putInCache(K key);
```
Comment (glmxndr) [inline common/common-utils/src/main/java/com/csgroup/s2pdgs/commonutils/DedupCache.java:14]: This action expicitly puts in cache.

File: common/common-utils/src/main/java/com/csgroup/s2pdgs/commonutils/DedupCache.java [hunk 7]
```diff
@@ -9,19 +9,33 @@
...
+    Mono<Boolean> isNotInCache(K key);
```
Comment (glmxndr) [inline common/common-utils/src/main/java/com/csgroup/s2pdgs/commonutils/DedupCache.java:15]: This action does not implicitly put in cache.

File: common/common-utils/src/main/java/com/csgroup/s2pdgs/commonutils/DedupCache.java [hunk 8]
```diff
@@ -9,19 +9,33 @@
...
+
+    default Mono<Boolean> isNotInCacheOrAddIt(K key) {
```
Comment (glmxndr) [inline common/common-utils/src/main/java/com/csgroup/s2pdgs/commonutils/DedupCache.java:17]: This action explicitly puts in cache. Bad name but meh.

File: common/common-utils/src/main/java/com/csgroup/s2pdgs/commonutils/DedupCache.java [hunk 9]
```diff
@@ -9,19 +9,33 @@
...
+        return isNotInCache(key).doOnNext(b -> putInCache(key));
+    }
+    default Mono<Boolean> isInCacheOrAddIt(K key) {
```
Comment (glmxndr) [inline common/common-utils/src/main/java/com/csgroup/s2pdgs/commonutils/DedupCache.java:20]: This action explicitly puts in cache. Bad name but meh.

File: applications/processor/session-filter/src/main/java/com/csgroup/s2pdgs/sessionfilter/SessionFilterProcessor.java [hunk 10]
```diff
@@ -149,7 +149,7 @@ private Session createSession(StationProduct dsib) {
         return catalogClientManager
             .monoWithCatalogClient(
                 catalog,
-                client -> client.addAndRefresh(catalogIndex, session.info())
+                client -> client.add(catalogIndex, session.info())
```
Comment (glmxndr) [inline applications/processor/session-filter/src/main/java/com/csgroup/s2pdgs/sessionfilter/SessionFilterProcessor.java:152]: Is this refresh really useful? It hurts a little.

File: applications/source/station-source/src/main/java/com/csgroup/s2pdgs/stationsource/StationProductsFinder.java [hunk 11]
```diff
@@ -148,15 +150,16 @@ public StationProductsFinder(
...
-            .doOnNext(p -> log.debug("readDSIB: {} -> {}", file.getName(), p.getSessionID()))
+            .doOnNext(p -> log.trace("readDSIB: {} -> {}", file.getName(), p.getSessionID()))
             .doOnError(t -> log.error("Exception reading DSIB file {}", file, t))
             .onErrorResume(t -> Mono.empty());
     }
     protected Mono<StationProduct> addToCatalog(StationProduct p){
         return clientManager.monoWithCatalogClient(
                 catalog,
-                    client -> client.addAndRefresh(catalogIndex, p)
+                    client -> client.add(catalogIndex, p)
```
Comment (glmxndr) [inline applications/source/station-source/src/main/java/com/csgroup/s2pdgs/stationsource/StationProductsFinder.java:160]: Is this refresh really useful? It hurts a LOT.

**Review by ilaborie — APPROVED** [review]

**Pull request merged**